test/unit/math/rev/mat/vectorize/apply_scalar_unary_test.cpp

```diff
@@ -0,0 +1,18 @@
+#include <stan/math/rev/core/var.hpp>
+#include <stan/math/rev/scal/fun/exp.hpp>
+#include <stan/math/rev/mat/vectorize/apply_scalar_unary.hpp>
+#include <test/unit/math/prim/mat/vectorize/foo_base_test.hpp>
+#include <test/unit/math/prim/mat/vectorize/expect_types.hpp>
+#include <test/unit/math/rev/mat/vectorize/expect_values.hpp>
+#include <test/unit/math/rev/mat/vectorize/expect_errors.hpp>
+#include <gtest/gtest.h>
+
+/**
+ * this tests that the expect_values test works on a mock function 
+ */
+TEST(MathRevMatVectorize, applyScalarUnaryMock) {
```

Comment:
This applies to all the `apply_scalar_unary_test.cpp` files.... I think you could get away with using a type-parameterized test (`TYPE_TEST_CASE_P`). https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md#type-parameterized-tests

It'll allow you to avoid rewriting this bit over and over again. If you want, we can discuss in person.